.github/workflows/run-k8s-tests-on-cloud-instance.yaml

```diff
@@ -0,0 +1,143 @@
+      KATA_HYPERVISOR: ${{ matrix.vmm }}
+      KUBERNETES: ${{ matrix.k8s }}
+      SNAPSHOTTER: ${{ matrix.snapshotter }}
+      USING_NFD: "false"
+      K8S_TEST_HOST_TYPE: ${{ matrix.instance-type }}
+      INSTANCE_NAME: ${{ matrix.instance }}
+    steps:
+      - name: Adjust a permission for repo
+        run: sudo chown -R $USER:$USER $GITHUB_WORKSPACE
+        if: matrix.instance == 's390x' && matrix.vmm == 'qemu'
+
+      - name: Take a pre-action for self-hosted runner
+        run: ${HOME}/script/pre_action.sh ubuntu-2204
+        if: matrix.instance == 's390x' && matrix.vmm == 'qemu'
+
+      - uses: actions/checkout@v4
+        with:
+          ref: ${{ inputs.commit-hash }}
+          fetch-depth: 0
+        if: |
+          matrix.instance == 'garm-ubuntu-2004' ||
+          matrix.instance == 'garm-ubuntu-2004-smaller' ||
+          matrix.vmm == 'qemu'
+
+      - name: Rebase atop of the latest target branch
+        run: |
+          ./tests/git-helper.sh "rebase-atop-of-the-latest-target-branch"
+        env:
+          TARGET_BRANCH: ${{ inputs.target-branch }}
+        if: |
+          matrix.instance == 'garm-ubuntu-2004' ||
+          matrix.instance == 'garm-ubuntu-2004-smaller' ||
+          matrix.vmm == 'qemu'
+
+      - name: Deploy ${{ matrix.k8s }}
+        run: bash tests/integration/kubernetes/gha-run.sh deploy-k8s
+        if: |
+          matrix.instance == 'garm-ubuntu-2004' ||
+          matrix.instance == 'garm-ubuntu-2004-smaller' ||
+          matrix.vmm == 'qemu'
+
+      - name: Configure the ${{ matrix.snapshotter }} snapshotter
+        run: bash tests/integration/kubernetes/gha-run.sh configure-snapshotter
+        if: |
+          matrix.instance == 'garm-ubuntu-2004' ||
+          matrix.instance == 'garm-ubuntu-2004-smaller' ||
+          matrix.vmm == 'qemu'
+
+      - name: Deploy Kata
+        timeout-minutes: 10
+        run: |
+        suffix=$(echo "$INSTANCE_NAME" | cut -d'-' -f 1)
+        bash tests/integration/kubernetes/gha-run.sh deploy-kata-${suffix}
+        if: |
+          matrix.instance == 'garm-ubuntu-2004' ||
+          matrix.instance == 'garm-ubuntu-2004-smaller' ||
+          matrix.vmm == 'qemu'
+
+      - name: Install `bats`
+        run: bash tests/integration/kubernetes/gha-run.sh install-bats
+        if: |
+          matrix.instance == 'garm-ubuntu-2004' ||
+          matrix.instance == 'garm-ubuntu-2004-smaller'
+  
+      - name: Run tests
+        timeout-minutes: 30
+        run: |
+        export TARGET_ARCH="$(uname -m)"
+        bash tests/integration/kubernetes/gha-run.sh run-tests
+        if: |
+          matrix.instance == 'garm-ubuntu-2004' ||
+          matrix.instance == 'garm-ubuntu-2004-smaller' ||
+          matrix.vmm == 'qemu'
+  
+      - name: Delete kata-deploy
+        run: |
+          suffix=$(echo "$INSTANCE_NAME" | cut -d'-' -f 1)
+          bash tests/integration/kubernetes/gha-run.sh cleanup-${suffix}
+        if: |
```

Comment:
the aforementioned conflict with `always()`

Reply:
So, we always want to delete kata-deploy, why would this be different for s390x?